packages/eslint-plugin/src/configs/restricted-globals.js

```diff
@@ -1,243 +1,240 @@
-// These will be enabled in a follow up PR.
-// https://github.com/microsoft/fluentui/issues/29964
-
 // Via: https://developer.mozilla.org/en-US/docs/Web/API/Window
-// const windowKeys = [
-//   // Instance properties
-//   'caches',
-//   'clientInformation',
-//   'closed',
-//   'console',
-//   'credentialless',
-//   'crypto',
-//   'customElements',
-//   'devicePixelRatio',
-//   'frameElement',
-//   'frames',
-//   'fullScreen',
-//   'history',
-//   'indexedDB',
-//   'innerHeight',
-//   'innerWidth',
-//   'isSecureContext',
-//   'launchQueue',
-//   'length',
-//   'localStorage',
-//   'location',
-//   'locationbar',
-//   'menubar',
-//   'name',
-//   'navigation',
-//   'navigator',
-//   'opener',
-//   'origin',
-//   'outerHeight',
-//   'outerWidth',
-//   'pageXOffset',
-//   'pageYOffset',
-//   'parent',
-//   'performance',
-//   'personalbar',
-//   'scheduler',
-//   'screen',
-//   'screenX',
-//   'screenY',
-//   'scrollbars',
-//   'scrollX',
-//   'scrollY',
-//   'self',
-//   'sessionStorage',
-//   'speechSynthesis',
-//   'statusbar',
-//   'toolbar',
-//   'top',
-//   'visualViewport',
-//   'window',
+const windowKeys = [
```

Comment:
this will become a maintenance burden, can we leverage [something that is fully automated](https://github.com/sindresorhus/globals) or bring the automation to our toolchain ?

Reply:
I like this idea! Added `globals` and it's already found more stuff :)

Reply:
I've run into an interesting situation here. `globals` includes things like `HTMLElement` and `KeyboardEvent`. Calling `new HTMLElement()` always throws so there's no real need to lint for it. However, you can call the `KeyboardEvent` constructor so we do want to lint for it.

The issue is that there are quite a few instances of `HTMLElement` used in our types and `eslint` is flagging them as using a restricted global. I don't think we want to add eslint exceptions for all our types. In the case of things like `HTMLElement` it's easy enough to just exclude them but `KeyboardEvent` cannot be excluded if we want an exhaustive rule.

@Hotell Are you aware of a way to exclude types from linting? I'd like to be able to write

```ts
const element: HTMLElement;
const event: KeyboardEvent;
```

And not

```ts
// eslint-disable-next-line no-restricted-globals
const element: HTMLElement;
// eslint-disable-next-line no-restricted-globals
const event: KeyboardEvent;
```

I was not able to find anything.

Reply:
I've been working at this a bit and have tried things like using [`ban-types`](https://typescript-eslint.io/rules/ban-types/), which can also be used to allow types, despite its name and I'm still hitting the issue outlined above.

I'm inclined to go back to maintaining our own list of disallowed globals but culling the list back to only things we know are issues like `setTimeout()`.

WDYT?

Reply:
Looks like we might need to make a type-aware extension to `no-restricted-globals` if we want to allow types but not values.

See: https://github.com/typescript-eslint/typescript-eslint/issues/4715

Reply:
> I'm inclined to go back to maintaining our own list of disallowed globals but culling the list back to only things we know are issues like setTimeout().

that sounds good to me for mitigating our initial core issues with using globals.

> Are you aware of a way to exclude types from linting? I'd like to be able to write

that's not something that can be done by some general config AFAIK, it needs to be implemented within the custom Lint rule, so it can omit checking TSESTree symbols that name starts with `TS*`.

Reply:
Made a follow up issue to look into alternatives: #31286